ssstar/src/objstore/s3.rs

```diff
@@ -1451,6 +1451,35 @@ impl MultipartUploader for S3MultipartUploader {
     }
 }
 
+#[derive(Debug)]
+pub struct UserAgentInterceptor {
+    user_agent: Option<HeaderValue>,
+}
+
+impl Interceptor for UserAgentInterceptor {
+    fn modify_before_signing(
+        &self,
+        context: &mut BeforeTransmitInterceptorContextMut,
+        _runtime_components: &RuntimeComponents,
+        _cfg: &mut ConfigBag,
+    ) -> Result<(), BoxError> {
+        let request = context.request_mut();
+
+        if let Some(ua) = self.user_agent.clone() {
+            const HEADER_NAME: &str = "user-agent";
+            request
+                .headers_mut()
+                .insert(HeaderName::from_static(HEADER_NAME), ua);
```

Comment:
HTTP headers are case-insensitive, but let's use the casing of the canonical representation of this header as specified in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent). We also don't need a `const` variable for that, let's simplify it.

```suggestion
            request
                .headers_mut()
                .insert(HeaderName::from_static("User-Agent"), ua);
```

Reply:
No, it's case sensitive in case with this particular function:
```
This function requires the static string to only contain lowercase characters, numerals and
symbols, as per the HTTP/2.0 specification and header names internal representation within this library.
```

Reply:
See: https://docs.rs/http/latest/http/header/struct.HeaderName.html#method.from_static